src/coreclr/jit/fgdiagnostic.cpp

```diff
@@ -2957,6 +2957,13 @@ void Compiler::fgDebugCheckBBlist(bool checkBBNum /* = false */, bool checkBBRef
 
         maxBBNum = max(maxBBNum, block->bbNum);
 
+        // BBJ_COND's normal (false) jump target is expected to be the next block
+        // TODO: Allow bbNormalJumpDest to diverge from bbNext
```

Comment:
A little tip for writing future `TODO`s: in ongoing work, it is handy to add a unique postfix to a `TODO`, e. g. `TODO-NoFallThrough`, s. t. they can be text-searched for easily afterwards.

Reply:
Thank you for the tip; I realized this might be a better approach about halfway through... I'll update the TODOs I introduced to make them easier to Ctrl+F when I remove them in the next PR.